File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXPanel.php
<?php
namespace php\gui\layout;

use php\gui\paint\UXColor;
use php\gui\text\UXFont;

/**
 * Class UXPanel
 * @package php\gui\layout
 * @packages gui, javafx
 */
class UXPanel extends UXAnchorPane
{
    /**
     * Заголовок
     * @var string
     */
    public $title;

    /**
     * Цвет заголовка.
     * @var UXColor
     */
    public $titleColor;

    /**
     * Шрифт заголовка.
     * @var UXFont
     */
    public $titleFont;

    /**
     * Позиция заголовка TOP_LEFT, TOP_RIGHT, TOP_CENTER.
     * @var string
     */
    public $titlePosition = 'TOP_LEFT';

    /**
     * Смещение заголовка.
     * @var int
     */
    public $titleOffset = 15;

    /**
     * @var int
     */
    public $borderWidth = 1;

    /**
     * @var UXColor
     */
    public $borderColor;

    /**
     * @var int
     */
    public $borderRadius = 0;

    /**
     * @var string SOLID, DOTTED, DASHED, NONE
     */
    public $borderStyle = 'SOLID';
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXPanel.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.geometry.Pos;
import javafx.scene.layout.AnchorPane;
import javafx.scene.paint.Color;
import javafx.scene.text.Font;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.support.control.Panel;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "layout\\UXPanel")
public class UXPanel<T extends Panel> extends UXAnchorPane<Panel> {
    interface WrappedInterface {
        @Property String title();
        @Property Color titleColor();
        @Property Font titleFont();
        @Property double titleOffset();
        @Property Pos titlePosition();

        @Property @Nullable Color borderColor();
        @Property double borderWidth();
        @Property double borderRadius();
        @Property String borderStyle();
    }

    public UXPanel(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXPanel(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public void __construct() {
        __wrappedObject = new Panel();
    }

    @Override
    @SuppressWarnings("unchecked")
    public T getWrappedObject() {
        return (T) super.getWrappedObject();
    }

    @Override
    @Reflection.Setter
    public void setBackgroundColor(@Nullable Color color) {
        getWrappedObject().setBackgroundColor(color);
    }

    @Override
    @Reflection.Getter
    public Color getBackgroundColor() {
        return getWrappedObject().getBackgroundColor();
    }
}
